.gitlab-ci.yml

```diff
@@ -110,3 +110,24 @@ build-wasm-spiritnet:
       - out/*.json
       - out/*.txt
     expire_in: 12 week
+
+build-wasm-try-runtime:
+  parallel:
+    matrix:
+      - RUNTIME: "peregrine"
+      - RUNTIME: "spiritnet"
+  image: paritytech/ci-unified:bullseye-1.70.0
+  stage: build
+  only:
+    - develop
+    - master
+    - /^[0-9]+(?:\.[0-9]+){2}(?:-(rc)*([0-9])+)?$/
+  script:
+    - cargo build --release -p $RUNTIME-runtime --features try-runtime
```

Comment:
`--locked`? Also, I see srtool has a `default-features` option which we could use to specify the `try-runtime` feature. I think it would not hurt if we would use srtool also to generate the WASM for runtime upgrade testing. What do you think?

Reply:
Yeah I thought about using the SR Tool and also make the 4 options (spiritnet, peregrine, spiritnet+try-runtime, peregrine+try-runtime) a matrix. But I didn't want to mix up the production build with the try-runtime builds. I feared that it would make it easier to mix them up.

Also why would you use the SR-tool for this?

Reply:
> Yeah I thought about using the SR Tool and also make the 4 options (spiritnet, peregrine, spiritnet+try-runtime, peregrine+try-runtime) a matrix. But I didn't want to mix up the production build with the try-runtime builds. I feared that it would make it easier to mix them up.

Agree. They can be two separate matrices.

> Also why would you use the SR-tool for this?

Since we enable SRTool-built runtimes, using SRTool also for try-runtime builds would make sure the code is consistent. We test the exact same code that will be deployed, without the `try-runtime` feature.

Reply:
Not sure how much we win. AFAIK we use the same compiler in the SR-Tool and the CI. The feature already messes up the wasm, so that it's not the same as the production version. Not sure how much the SR-Tool will help there. But I guess it's really doesn't hurt. 🤷

Reply:
Yes, probably not a big win overall, but if the effort is not too big, I would go down this road. Also from a mental model point of view, we are now making sure the SRTool is used to build all WASM artifacts, regardless of their features. If SRTool does not make it possible to build the try-runtime WASM, then of course the CI is close enough, I guess.